examples/summarization/bart/run_bart_sum.py

```diff
@@ -0,0 +1,247 @@
+            optimizer, num_warmup_steps=self.hparams.warmup_steps, num_training_steps=t_total
+        )
+        self.lr_scheduler = scheduler
+        return [optimizer]  # , [scheduler]
+
+    def prepare_data(self):
+        self.train_dataset = CnnDailyMailDataset(self.tokenizer, data_dir=self.hparams.data_dir)
+        self.val_dataset = CnnDailyMailDataset(self.tokenizer, data_dir=self.hparams.data_dir, type_path="val")
+        # self.test_dataset = CnnDailyMailDataset(self.tokenizer, data_dir=self.hparams.data_dir, type_path='test')
+
+    def train_dataloader(self):
+        return DataLoader(self.train_dataset, batch_size=self.hparams.train_batch_size)
+
+    def val_dataloader(self):
+        return DataLoader(self.val_dataset, batch_size=self.hparams.eval_batch_size)
+
+    # def test_dataloader(self):
+    #     return DataLoader(self.test_dataset, batch_size=self.hparams.eval_batch_size)
+
+    @staticmethod
+    def add_model_specific_args(parser, root_dir):
+        # Add BART specific options
+        parser.add_argument(
+            "--max_seq_length",
+            default=128,
+            type=int,
+            help="The maximum total input sequence length after tokenization. Sequences longer "
+            "than this will be truncated, sequences shorter will be padded.",
+        )
+
+        parser.add_argument(
+            "--data_dir",
+            default=None,
+            type=str,
+            required=True,
+            help="The input data dir. Should contain the training files for the CoNLL-2003 NER task.",
+        )
+
+        parser.add_argument(
+            "--batch_size", default=4, type=int, help="The size of each batch.",
+        )
+
+        parser.add_argument(
+            "--config_name", default="", type=str, help="Pretrained config name or path if not the same as model_name"
+        )
+        parser.add_argument(
+            "--tokenizer_name",
+            default="",
+            type=str,
+            help="Pretrained tokenizer name or path if not the same as model_name",
+        )
+        parser.add_argument(
+            "--cache_dir",
+            default="",
+            type=str,
+            help="Where do you want to store the pre-trained models downloaded from s3",
+        )
+
+        parser.add_argument("--learning_rate", default=3e-5, type=float, help="The initial learning rate for Adam.")
+        parser.add_argument("--weight_decay", default=0.0, type=float, help="Weight decay if we apply some.")
+        parser.add_argument("--adam_epsilon", default=1e-8, type=float, help="Epsilon for Adam optimizer.")
+        parser.add_argument("--warmup_steps", default=0, type=int, help="Linear warmup over warmup_steps.")
+        parser.add_argument(
+            "--num_train_epochs", default=1, type=int, help="Total number of training epochs to perform."
+        )
+
+        parser.add_argument("--train_batch_size", default=4, type=int)
```

Comment:
already have batch_size (I know you didn't delete unused args yet)

Reply:
correct, removed batch_size